contributing/CODING_STANDARDS.md

```diff
@@ -92,51 +46,67 @@ Sometimes the test is trivial, in that the unit is small and only has
 one code path. In that case shoehorning a test case description into a
 `when`/`then` style may be less readable than a simpler ad-hoc
 alternative. Best judgement should be used, bearing in mind readability
-and consistency tradeoffs.
+and consistency trade-offs.
 
-### IDE configuration
+## C++
 
-To aid in conforming to our coding style a [`.editorconfig`](https://editorconfig.org/)
-file is included in the root of the repository, which has wide support
-across various IDEs. For supported IDEs this config file can override
-the default settings for inspections and formatting refactors.
+### Classes
 
-In particular, more granular settings are included for IntelliJ-based
-IDEs (for example, PyCharm) via the `ij_*` extensions.
+C++ classes that represent system components with reference semantics
+(as opposed to 'value' types) should inherit
+`std::enable_shared_from_this` and define a peer `Ptr` alias using
+the `openassetio::SharedPtr`.
 
-Common file types that we expect to encounter during development of
-OpenAssetIO have their own settings, many of which remain commented out
-for reference and may be enabled and tweaked in future.
+```cpp
+class MyClass : std::enable_shared_from_this<MyClass> { ... };
+using MyClassPtr = openassetio::SharedPtr<MyClass>;
+```
 
-As mentioned in the [C++](#c) section above, a `.clang-format` 
-configuration for C++ is included at the root of the repository, which
-has wide support in various IDEs to aid in auto-formatting.
+This is to simplify memory management across the complex range of
+language bindings within the project.
 
-### Environment variables
+## C
 
-All environment variables should be prefixed with `OPENASSETIO_`.
-For example, `OPENASSETIO_LOGGING_SEVERITY`.
+### Handles
 
-When documenting environment variables in docstrings or doxygen comment
-blocks, precede the variable name with the `@envvar` tag, which will
-cause the variable and its description to be listed in the _Environment
-Variable List_ page of the generated documentation.
+C Handles that manage reference semantics objects should always use a
+`<T>Ptr` over a raw pointer, and use a `Shared` suffix in their naming,
+to facilitate object exchange through multiple languages bindings.
 
-### Parameter documentation
+```cpp
+using SharedMyClass = Converter<MyClassPtr, oa_SharedMyClass_h>;
+```
 
-All method parameters should be documented in docstrings or doxygen
-comment blocks.
+## Python bindings
 
-In Python, as our configuration can't determine parameter types, we must
-manually specify those. This should be in the following form for
-OpenAssetIO defined types:
+### Holder classes for reference-semantic types
 
-```
-@param <name> <path> "<short.path>" Description text.
+When binding something with reference semantics to python, the holder
+should always be a `<T>Ptr`. `pybind` understands `std::shared_ptr`, and
```

Comment:
[consider] Probably good to use our alias
```suggestion
should always be a `<T>Ptr`. `pybind` understands `SharedPtr`, and
```

Reply:
Based on [the notes here](https://github.com/OpenAssetIO/OpenAssetIO/pull/443#discussion_r896746064), sticking with current form.